api/firmware/device.go

```diff
@@ -90,6 +90,14 @@ type Device struct {
 	channelHashAppVerified    bool
 	channelHashDeviceVerified bool
 	sendCipher, receiveCipher *noise.CipherState
+	// encrypting (decrypting) increments the cipherstate nonce, and has to be in sync with the
+	// state nonce in the BitBox02.
+	//
+	// We have to make sure that if there are concurrent queries, the incrementing of the local
+	// nonces must match the incrementing of the device nonces, so the decryption works. Avoid the
+	// situation where we Encrypt() locally twice, and send the queries out of order, which will
+	// result in a failed decryption on the device.
+	queryLock sync.Mutex
```

Comment:
> unless async rust could interfere somehow?

Not right now, as we only have one task that can be spawned at once, and hww.c can only do one query at a time (and responds with busy otherwise).

Reply:
Yeah, I was mostly concerned with the future. Need to remember this somehow.